src/utils/wallet.js

```diff
@@ -205,21 +320,34 @@ class Wallet {
             meta: { type }
         }) => permission === 'FullAccess' && type !== 'ledger' && !(localAccessKey && public_key === localAccessKey.public_key))
 
+        const WALLET_METADATA_METHOD = '__wallet__metadata'
+        let newLocalKeyPair
+        if (!localAccessKey || (!localAccessKey.access_key.permission.FunctionCall ||
+            !localAccessKey.access_key.permission.FunctionCall.method_names.includes(WALLET_METADATA_METHOD))) {
+            // NOTE: This key isn't used to call actual contract method, just used to verify connection with account in private DB
+            newLocalKeyPair = KeyPair.fromRandom('ed25519')
+            await account.addKey(newLocalKeyPair.getPublicKey(), this.accountId, WALLET_METADATA_METHOD, '0')
+        }
+
         for (const { public_key } of keysToRemove) {
             await account.deleteKey(public_key)
         }
 
-        let newLocalKeyPair = await this.addWalletMetadataAccessKeyIfNeeded(this.accountId, localAccessKey)
         if (newLocalKeyPair) {
             if (localAccessKey) {
                 await account.deleteKey(localAccessKey.public_key)
             }
             await this.keyStore.setKey(NETWORK_ID, this.accountId, newLocalKeyPair)
         }
 
-        const { data: recoveryMethods } =  await this.getRecoveryMethods();
+        const { data: recoveryMethods } = await this.getRecoveryMethods();
         for (const recoveryMethod of recoveryMethods) {
-            this.deleteRecoveryMethod(recoveryMethod)
+            try {
+                await this.deleteRecoveryMethod(recoveryMethod)
+            } catch(e) {
+                console.log(e)
```

Comment:
I don't think we should ignore error here. Makes sense to return error to caller.